src/symbol.rs

```diff
@@ -212,29 +210,29 @@ pub fn sysv_hash(s: &String) -> u64 {
 }
 
 
-pub fn symhash(eh: &Header, symbols: &Vec<Symbol>, link: u32) -> Section {
+pub fn symhash(eh: &Header, symbols: &Vec<Symbol>, link: u32) -> Result<Section, Error> {
     assert!(symbols.len() > 0);
     //TODO i'm too lazy to do this correctly now, so we'll just emit a hashtable with nbuckets  == 1
     let mut b = Vec::new();
     {
-        let mut io = &mut b;
-        elf_write_uclass!(eh, io, 1); //nbuckets
-        elf_write_uclass!(eh, io, symbols.len() as u64); //nchains
+        let io = &mut b;
+        try!(elf_write_uclass!(eh, io, 1)); //nbuckets
```

Comment:
is there a reason for try! versus ?, i prefer ? because error handling isn't interesting to the casual reader.  unless there's some rust coding guideline that says something else.